test/Dapr.Client.Test/StateApiTest.cs

```diff
@@ -224,6 +224,135 @@ public async Task SetStateAsync_ThrowsForNonSuccess()
             await FluentActions.Awaiting(async () => await task).Should().ThrowAsync<RpcException>();
         }
 
+        [Fact]
+        public async Task ExecuteStateTransactionAsync_CanSaveState()
+        {
+            // Configure Client
+            var httpClient = new TestHttpClient();
+            var daprClient = new DaprClientBuilder()
+                .UseGrpcChannelOptions(new GrpcChannelOptions { HttpClient = httpClient })
+                .Build();
+
+            var stateValue1 = new Widget() { Size = "small", Color = "yellow", };
+            var metadata1 = new Dictionary<string, string>()
+            {
+                {"a", "b" }
+            };
+            var options1 = new StateOptions();
+            options1.Concurrency = ConcurrencyMode.LastWrite;
+            ;
+            var state1 = new StateTransactionRequest("stateKey1", JsonSerializer.SerializeToUtf8Bytes(stateValue1), StateOperationType.Upsert, "testEtag", metadata1, options1);
+
+            int stateValue2 = 100;
+            var state2 = new StateTransactionRequest("stateKey2", JsonSerializer.SerializeToUtf8Bytes(stateValue2), StateOperationType.Delete);
+
+            string stateValue3 = "teststring";
+            var state3 = new StateTransactionRequest("stateKey3", JsonSerializer.SerializeToUtf8Bytes(stateValue3), StateOperationType.Upsert);
+
+            var states = new List<StateTransactionRequest>();
+            states.Add(state1);
+            states.Add(state2);
+            states.Add(state3);
+
+            var task = daprClient.ExecuteStateTransactionAsync("testStore", states);
+
+            // Get Request and validate
+            httpClient.Requests.TryDequeue(out var entry).Should().BeTrue();
+            var transactionRequest = await GrpcUtils.GetRequestFromRequestMessageAsync<Autogenerated.ExecuteStateTransactionRequest>(entry.Request);
+
+            transactionRequest.StoreName.Should().Be("testStore");
+            transactionRequest.Operations.Count.Should().Be(3);
+
+            var req1 = transactionRequest.Operations[0];
+            req1.Request.Key.Should().Be("stateKey1");
+            req1.OperationType.Should().Be(StateOperationType.Upsert.ToString().ToLower());
+            var valueJson1 = req1.Request.Value.ToStringUtf8();
+            var value1 = JsonSerializer.Deserialize<Widget>(valueJson1);
+            value1.Size.Should().Be(stateValue1.Size);
+            value1.Color.Should().Be(stateValue1.Color);
+            req1.Request.Etag.Should().Be("testEtag");
+            req1.Request.Metadata.Count.Should().Be(1);
+            req1.Request.Metadata["a"].Should().Be("b");
+            //optionsFound1.Should().Be(Autogenerated.StateOptions.Concurrency.FirstWrite);
```

Comment:
do you mean to have this commented out? Is this supposed to be here or not :)

Reply:
removed. I changed the data type so not needed